google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/GfeLatencyTest.java

```diff
@@ -0,0 +1,410 @@
+    long count1 =
+        getOverriddenHeaderMissingCount(
+            SpannerRpcViews.SPANNER_GFE_HEADER_MISSING_COUNT_VIEW,
+            projectId,
+            instanceId,
+            databaseId,
+            "google.spanner.v1.Spanner/ExecuteSql");
+    assertEquals(1, count1);
+  }
+
+  private static SpannerOptions createSpannerOptions(InetSocketAddress address, Server server) {
+    String endpoint = address.getHostString() + ":" + server.getPort();
+    return SpannerOptions.newBuilder()
+        .setProjectId("[PROJECT]")
+        // Set a custom channel configurator to allow http instead of https.
+        .setChannelConfigurator(
+            input -> {
+              input.usePlaintext();
+              return input;
+            })
+        .setHost("http://" + endpoint)
+        // Set static credentials that will return the static OAuth test token.
+        .setCredentials(STATIC_CREDENTIALS)
+        // Also set a CallCredentialsProvider. These credentials should take precedence above
+        // the static credentials.
+        .setCallCredentialsProvider(() -> MoreCallCredentials.from(VARIABLE_CREDENTIALS))
+        .build();
+  }
+
+  private long getAggregationValueAsLong(AggregationData aggregationData) {
+    return aggregationData.match(
+        new io.opencensus.common.Function<AggregationData.SumDataDouble, Long>() {
+          @Override
+          public Long apply(AggregationData.SumDataDouble arg) {
+            return (long) arg.getSum();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData.SumDataLong, Long>() {
+          @Override
+          public Long apply(AggregationData.SumDataLong arg) {
+            return arg.getSum();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData.CountData, Long>() {
+          @Override
+          public Long apply(AggregationData.CountData arg) {
+            return arg.getCount();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData.DistributionData, Long>() {
+          @Override
+          public Long apply(AggregationData.DistributionData arg) {
+            return (long) arg.getMean();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData.LastValueDataDouble, Long>() {
+          @Override
+          public Long apply(AggregationData.LastValueDataDouble arg) {
+            return (long) arg.getLastValue();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData.LastValueDataLong, Long>() {
+          @Override
+          public Long apply(AggregationData.LastValueDataLong arg) {
+            return arg.getLastValue();
+          }
+        },
+        new io.opencensus.common.Function<AggregationData, Long>() {
+          @Override
+          public Long apply(AggregationData arg) {
+            throw new UnsupportedOperationException();
+          }
+        });
+  }
+
+  private long getMetric(
```

Comment:
Could we merge the `getMetric` / `getOverriddenHeaderMissingCount` methods? They look pretty similar